.github/workflows/helm-chart-ci.yaml

```diff
@@ -165,39 +144,79 @@ jobs:
           verbosity: 1
 
+      - name: Setup Test dependencies
+        run: ./pre-install.sh
+        working-directory: .github/tests
+
       - name: Run chart-testing (install)
         run: |
-          post-install() {
-            [ -x "${TEST_DIR}/post-install.sh" ] && "${TEST_DIR}/post-install.sh" $1
-            exit $1
-          }
-
-          trap 'post-install $? $LINENO' EXIT
-
-          export scenario="$(basename "${TEST_DIR}")"
-          export EXTRA_HELM_ARGS=""
-
-          source .github/tests/charts.sh
-
-          [ "${scenario}" != "default" ] && kubectl create namespace "${scenario}"
-          [ -x "${TEST_DIR}/pre-install.sh" ] && "${TEST_DIR}/pre-install.sh"
-          [ -f "${TEST_DIR}/.env" ] && source "${TEST_DIR}/.env"
-
-          if [ -x "${TEST_DIR}/install.sh" ]; then
-            "${TEST_DIR}/install.sh"
-          else
-            ct install --debug \
-              --charts "charts/spire" \
-              --namespace "${scenario}" \
-              --target-branch ${{ github.base_ref }} \
-              --exclude-deprecated \
-              --skip-clean-up \
-              ${{ (matrix.values != 'default' && '--helm-extra-set-args "--values=${TEST_DIR}/values.yaml ${EXTRA_HELM_ARGS}"') || '' }}
-          fi
-        env:
-          TEST_DIR: .github/tests/${{ matrix.values }}
+          ct install --config ct.yaml \
+            --target-branch ${{ github.base_ref }}
+
+      - name: Test summary
+        if: always()
+        run: ./post-install.sh
+        working-directory: .github/tests
+
+  build-matrix:
+    name: Build matrix
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.5.2
+
+      - id: set-matrix
+        name: Collect all examples
+        run: |
+          examples="$(find examples -maxdepth 2 -type f -name run-tests.sh | xargs -I % dirname %)"
+          examples_json="$(echo "$examples" | jq -c --slurp --raw-input 'split("\n") | map(select(. != ""))')"
+          echo "${examples_json}"
+          echo "examples=$examples_json" >>"$GITHUB_OUTPUT"
+
+    outputs:
+      examples: ${{ steps.set-matrix.outputs.examples }}
+
+  example-test:
+    runs-on: ubuntu-22.04
+
+    needs:
+      - lint-chart
+      - build-matrix
+
+    strategy:
+      fail-fast: false
+      matrix:
+        example:
+          - ${{ fromJson(needs.build-matrix.outputs.examples) }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.5.2
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3.5
+        with:
+          version: ${{ env.HELM_VERSION }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.7.0
+        # Only build a kind cluster if there are chart changes to test.
+        with:
+          version: ${{ env.KIND_VERSION }}
+          node_image: kindest/node:v1.26.4
```

Comment:
as this workflow uses one version I prefer to keep it here to not abstract things to much and make it less transparant. This version is used only in a single place. Once we would use it in multiple places I would agree and move it to an env var probably. For now that seems like pre optimization, causing more complexity for users to hop through another abstraction to figure things out.

The reasoning to run on 1.26.4 is because that is the current `stable` version. All the other tests run on latest greatest release.

Reply:
It would be just to keep all the versions together in the config. It would be cleaner that way imo. But we can skip it too. Its not that big a deal.